### DISCUSSION TITLE
A Worrying Analysis of Probabilistic Time-series Models for Sales Forecasting

### DISCUSSION THREAD

**Discussion by StatMixedML** [awslabs/gluonts#1180] [General]
I want to share this paper with you: 

[A Worrying Analysis of Probabilistic Time-series Models for Sales Forecasting](https://arxiv.org/abs/2011.10715)

The authors compare the performance of probabilistic time series models (e.g., DeepAR, DeepState) to simpler models, like MLP or LR. In their experimental results, the authors show **that a simple Multi-layer Perceptron and Linear Regression outperform the probabilistic models on RMSE**.

I guess it is heavily depending on the evaluation setup and on the data set used. Anyway, I am curious to hear your thoughts.

**Comment by timoschowski** [comment]
Here's a thought:
https://www.sciencedirect.com/science/article/abs/pii/S0169207019301359

**Comment by StatMixedML** [comment]
@timoschowski Thanks for opening the discussion.

I do agree that the choice of an appropriate evaluation metric is crucial and, depending on the metric chosen, results and conclusions drawn can be quite different. It is also important to note that there is no "one beats all model" that outperforms any other model. There are instances where a simple model outperforms more sophisticated ones, and vice versa, of course. Also, I am not sure to what extent a probabilistic model should be evaluated using point forecasts. But that is a different story altogether...

I do want, however, to highlight one interesting point made in the paper: it is on the data and its characteristics on which the models have been evaluated on. The following figure shows a comparison of spectral entropy density for the datasets used. The figure shows the "forecastability" for EC dataset and publicly available datasets, where the spectral entropy measure is defined as a value between 0% and 100%, where 100 means not forecastable (white noise) and 0 means perfectly forecastable.

![image](https://user-images.githubusercontent.com/41187941/100613996-b6416180-3315-11eb-8185-70bfa7644acf.png)

What is obvious is that the spectral entropy densities of widely used public datasets such as trafﬁc, electricity, and Wikipedia are biased towards a speciﬁc point. In contrast, the entropy of the EC dataset is much more distributed than that of others. It would be interesting then to see how probabilistic models perform on more spectral entropy dispersed data sets.

**Comment by jgasthaus** [comment]
Interesting paper, thanks for sharing! It's a bit unfortunate that it uses a non-public data set as well as custom implementations, making it hard to reproduce and investigate these results further.

> The figure shows the "forecastability" for EC dataset and publicly available datasets, where the spectral entropy measure is defined as a value between 0% and 100%, where 0 means not forecastable (white noise) and 100 means perfectly forecastable.

I think it's actually the other way around -- lower entropy -> more forecastable. This aligns well with my experience: the traffic and electricity data sets are fairly "easy", in that they mostly have stable seasonality and little noise. A seasonal naive model performs almost as well as DeepAR or other deep models on these data sets, and without additional covariates, I suspect it's almost impossible to perform significantly better in terms of point predictions.

**Comment by vafl** [comment]
Thanks a lot for the paper! This is indeed interesting. In addition to what has been said above: Both accuracy measures MAPE and RMSE are dominated by the time series with the largest magnitude, in particular, if the magnitude follows a power law in the dataset as is the case for e.g. wikipedia.
So it may be that most of the error comes from a handful of time series. Also the time series with largest magnitude are often the most regular and seasonal time series. 
Still the result is interesting, and I think we have seen also in other experiments that MLPs can be quite competitive.

**Comment by StatMixedML** [comment]
I also want to draw your attention to this workshop

[I Can't Believe It's Not Better! Workshop, ICBINB@NeurIPS 2020](https://i-cant-believe-its-not-better.github.io/)

where the above mentioned paper has been accepted. I cannot agree more to [this:](https://i-cant-believe-its-not-better.github.io/impact/)

> Understanding counter-intuitive results is essential, but under-valued, in an increasingly competitive and fast-paced research environment. This workshop aims to rescue papers with fresh and powerful insights that would otherwise fall into the graveyard of forgotten papers. Such works might be high-hanging fruits in the tree of science, and are often those that need help the most.

**Comment by jgasthaus** [comment]
> @jgasthaus
> 
> > I think it's actually the other way around
> 
> The authors use [Forecastable Component Analysis](http://proceedings.mlr.press/v28/goerg13.pdf) for calculating the entropy measure. According to the [manual](https://github.com/gmgeorg/ForeCA/blob/master/R/Omega.R), it says:
> 
> _A real-value between 0 and 100 %. 0 means not forecastable (white noise); 100 means perfectly forecastable (a sinusoid)_.

The forecastability measure \Omega in the "Forecastable Component Analysis" paper is defined as one minus (scaled) spectral entropy, but the authors here plot the spectral entropy directly. 

They also explicitly state that 
"Figure 1 shows the comparison of widely used public datasets and the EC dataset in terms of forecastability. Our dataset is more trickier than traffic and electricity but easier than Wikipedia."

**Comment by StatMixedML** [comment]
@jgasthaus thanks for clarifying! I have also changed my description above.
